common/src/main/java/org/opensearch/ml/common/model/LocalRegexGuardrail.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.model;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.action.LatchedActionListener;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.common.CommonValue.stopWordsIndices;
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+@Log4j2
+@EqualsAndHashCode
+@Getter
+public class LocalRegexGuardrail extends Guardrail {
+    public static final String STOP_WORDS_FIELD = "stop_words";
+    public static final String REGEX_FIELD = "regex";
+
+    private List<StopWords> stopWords;
+    private String[] regex;
+    private List<Pattern> regexPattern;
+    private Map<String, List<String>> stopWordsIndicesInput;
+    private NamedXContentRegistry xContentRegistry;
+    private Client client;
+
+    @Builder(toBuilder = true)
+    public LocalRegexGuardrail(List<StopWords> stopWords, String[] regex) {
+        this.stopWords = stopWords;
+        this.regex = regex;
+    }
+    public LocalRegexGuardrail(@NonNull Map<String, Object> params) {
+        Object words = params.get(STOP_WORDS_FIELD);
+        stopWords = new ArrayList<>();
+        for (Map e : (List<Map>)words) {
+            stopWords.add(new StopWords(e));
+        }
+        this.regex = ((List<String>) params.get(REGEX_FIELD)).toArray(new String[0]);
```

Comment:
Add null check for `params.get(REGEX_FIELD)` and `params.get(STOP_WORDS_FIELD)` ?

Reply:
done.

Reply:
Added UT.